app/controllers/application_controller.rb

```diff
@@ -16,4 +16,10 @@ def authorize!
       redirect_to root_path
     end
   end
+
+  def authorize_company!
+    unless (current_user && (current_user.admin? || current_user.company.id.to_s == params[:id]))
+      redirect_to root_path
+    end
```

Comment:
Great! I like that it checks that company you are associated with and compares that to the company page you are trying to visit.